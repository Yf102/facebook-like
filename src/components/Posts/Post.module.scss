.post-container {
  @apply m-4 lg:m-8 px-5 py-8 flex flex-col;
  gap: 1.5rem;
  background-color: var(--light-gray);

  .img-wrapper {
    @apply relative w-full;
    height: 0;
    padding-bottom: 56.25%;
    overflow: hidden;
    box-shadow: 0 0 0 1px rgba(0, 0, 0, 0.05) inset;
    background-color: var(--light-gray);
  }
}