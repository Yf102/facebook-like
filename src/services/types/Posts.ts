type StatusType<T> = {
  status_code: number;
  status: string;
  data: T;
};

export type PostsApiType = StatusType<Posts>;

export type Posts = {
  page: number;
  posts: Post[];
};

export type Post = {
  id: number;
  title: string;
  image?: string;
  content: string;
};

export type PostsParamsType = {
  page: number;
};
