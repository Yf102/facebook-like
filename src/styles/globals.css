@tailwind base;
@tailwind components;
@tailwind utilities;

html,
body, #root {
  margin: 0;
  padding: 0;
  height: 100%;
  width: 100%;
}

h2 {
  color:#222;
  font-weight:600;
  margin: 0;
}

:root {
  --foreground-rgb: 0, 0, 0;
  --background-start-rgb: 214, 219, 220;
  --background-end-rgb: 255, 255, 255;
  --light-gray: #EFEFEF
}

@media (prefers-color-scheme: dark) {
  :root {
    --foreground-rgb: 255, 255, 255;
    --background-start-rgb: 0, 0, 0;
    --background-end-rgb: 0, 0, 0;
  }
}

body {
  background-color: #D9D9D9;
  color: rgb(var(--foreground-rgb));
}
