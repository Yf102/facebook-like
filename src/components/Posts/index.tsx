import React from "react";
import styles from "./Post.module.scss";
import Image from "next/image";
import { Post } from "@/src/services/types/Posts";

type PostsProps = { posts: Post[] };
const Posts = ({ posts }: PostsProps) => {
  return (
    <>
      {posts.map(({ id, title, image, content }) => {
        return (
          <div key={id} className={styles["post-container"]}>
            <h2>{title}</h2>
            {image && (
              <div className={styles["img-wrapper"]}>
                <Image src={image} alt={title} fill />
              </div>
            )}
            <div>{content}</div>
          </div>
        );
      })}
    </>
  );
};

export default Posts;
