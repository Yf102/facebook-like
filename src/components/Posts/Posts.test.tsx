import React from "react";
import { render } from "@/tests/test-utils";
import Posts from "@/src/components/Posts/index";
import { describe, expect, it } from "@jest/globals";
import { postsData } from "@/tests/__mocks__/data/Posts";

describe("Posts", () => {
  it("It renders multiple posts", async () => {
    const { asFragment } = render(<Posts posts={postsData} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
