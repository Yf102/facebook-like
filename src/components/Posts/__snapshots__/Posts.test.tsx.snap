// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts It renders multiple posts 1`] = `
<DocumentFragment>
  <div
    class="post-container"
  >
    <h2>
      Testing title 1
    </h2>
    <div
      class="img-wrapper"
    >
      <img
        alt="Testing title 1"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="100vw"
        src="/_next/image?url=https%3A%2F%2Fpicsum.photos%2Fid%2F208%2F854%2F480&w=3840&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fpicsum.photos%2Fid%2F208%2F854%2F480&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fid%2F208%2F854%2F480&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fid%2F208%2F854%2F480&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fid%2F208%2F854%2F480&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fid%2F208%2F854%2F480&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fid%2F208%2F854%2F480&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fid%2F208%2F854%2F480&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fid%2F208%2F854%2F480&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
    </div>
    <div>
      Testing content 1
    </div>
  </div>
  <div
    class="post-container"
  >
    <h2>
      Testing title 3
    </h2>
    <div>
      Testing content 3
    </div>
  </div>
</DocumentFragment>
`;
